packages/dev/serializers/src/glTF/2.0/glTFMaterialExporter.ts

```diff
@@ -373,6 +211,11 @@ export class _GLTFMaterialExporter {
                     })
                 );
             }
+
+            if (promises.length > 0) {
+                this._exporter._materialNeedsUVsSet.add(babylonStandardMaterial);
+                await Promise.all(promises);

```

Comment:
With these parallel promises, is the order of the materials in the exported file deterministic?

Reply:
@alexchuber

Reply:
I'm not sure I understand this question fully, so let me know if I've missed something.

In the "materials" glTF array, the order of materials is deterministic, but only because of some `await`s in `_exportMaterialAsync` (that I'm not fully sure need to be there in the first place.) I was planning on revisiting this another time to see it if can be more parallelized.

As for the promise chain of textures here... I think they appear in the glTF material object in the order defined by `IMaterial` in glTF2Interface.ts.

Reply:
It is deterministic because the async part of `_exportTextureAsync` is at the end when it needs get the image data asynchronously.

Reply:
I'm not sure what you are asking. The order is deterministic if the scene doesn't change. If the scene changes, then there is a possibility the order will also change since we traverse the scene to export the materials.